### PR TITLE
handle semantic version numbers in interface/sandbox update

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10492/HighFidelity-Beta-PR13344-7a9d77d.apk

**Comment by birarda** [comment]
Test Plan

##### PR Build
Run the Interface and Sandbox from this PR. Confirm that the version reported is PR13344-7a9d77d and that neither the Interface or the Sandbox prompts you for any available update.

##### Newer Master Build
Run Interface and Sandbox for this fake master build for 8500 - . Confirm that the version reported is 8500-7a9d77d and that neither the Interface or the Sandbox prompts you for any available update.

##### Older Master Build
Run Interface and Sandbox for this fake master build for 8300 - . Confirm that the version reported is 8300-7a9d77d and that you are prompted to update to the latest version (which should be higher than 8385) by both Interface and the Sandbox. Confirm that you see the release notes for versions later than 8300 in Interface. Confirm that in both Interface and the Sandbox clicking the link to update downloads the build with the version number displayed in the update dialog.

##### Newer Stable Build
Run Interface and Sandbox for this fake stable build for 0.69.0 - . Confirm that the version reported is 0.69.0 and that you are not prompted to update by either the Interface or the Sandbox.

##### Older Stable Build
Run Interface and Sandbox for this fake stable build for 0.68.0 - . Confirm that the version reported is 0.68.0 and that you are prompted to update to the latest version (which should be 0.69.0) by both Interface and the Sandbox. Confirm that you see release notes for the test version 0.69.0 in Interface. Confirm that in both Interface and the Sandbox clicking the link to update sends you to `https://builds.highfidelity.com/HighFidelity-Beta-0.69.0.exe` (which is a 404).

**Comment by birarda** [comment]
This will not be QA ready until I post the links for the faked builds for testing. I will mark it as such when that is done.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10492/HighFidelity-Beta-PR13344-7a9d77d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10492/HighFidelity-Beta-Interface-PR13344-7a9d77d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10492/HighFidelity-Beta-PR13344-7a9d77d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10492/HighFidelity-Beta-Interface-PR13344-7a9d77d.exe

**Comment by Qliemillar** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10543/HighFidelity-Beta-PR13344-7a9d77d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10543/HighFidelity-Beta-PR13344-7a9d77d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10543/HighFidelity-Beta-Interface-PR13344-7a9d77d.dmg

**Comment by qlimguicheteau** [comment]
Now Testing

**Comment by qlimguicheteau** [comment]
FAIL @conklin94122 

The update numbers are correct, but the sandbox does not trigger an update notice.
The only notices were directly inside the Interface.
update2

The first update link goes to build 8412.
The second link leads to https://builds.highfidelity.com/HighFidelity-Beta-5654.exe instead of a 404 for the 69 build.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10546/HighFidelity-Beta-PR13344-4713d8d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10546/HighFidelity-Beta-PR13344-4713d8d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10546/HighFidelity-Beta-Interface-PR13344-4713d8d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10546/HighFidelity-Beta-PR13344-4713d8d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10546/HighFidelity-Beta-Interface-PR13344-4713d8d.exe

File: interface/src/ui/UpdateDialog.cpp [hunk 1]
```diff
@@ -21,19 +21,31 @@ UpdateDialog::UpdateDialog(QQuickItem* parent) :
...
-    _releaseNotes = "";
-    for (int i = latestVersion; i > currentVersion; i--) {
-        if (applicationUpdater.data()->getBuildData().contains(i)) {
-            QString releaseNotes = applicationUpdater.data()->getBuildData()[i]["releaseNotes"];
-            releaseNotes.remove("<br />");
-            releaseNotes.remove(QRegExp("^\n+"));
-            _releaseNotes += "\n" + QString().sprintf("%d", i) + "\n" + releaseNotes + "\n";
+    if (applicationUpdater) {
+
+        auto buildData = applicationUpdater.data()->getBuildData();
```
Comment (Atlante45) [inline interface/src/ui/UpdateDialog.cpp]: You can just do `applicationUpdater->getBuildData()`

File: interface/src/ui/UpdateDialog.cpp [hunk 2]
```diff
@@ -47,5 +59,5 @@ const QString& UpdateDialog::releaseNotes() const {
 
 void UpdateDialog::triggerUpgrade() {
     auto applicationUpdater = DependencyManager::get<AutoUpdater>();
-    applicationUpdater.data()->performAutoUpdate(applicationUpdater.data()->getBuildData().lastKey());
+    applicationUpdater.data()->openLatestUpdateURL();
```
Comment (Atlante45) [inline interface/src/ui/UpdateDialog.cpp]: You can just do `applicationUpdater-> openLatestUpdateURL()`

**Comment by qlimguicheteau** [comment]
Now Testing

**Comment by qlimguicheteau** [comment]
FAIL @conklin94122 

All files downloaded had their version SHA match the number on the link. (this is different from the written test numbers, but assumed pass because the link numbers were changed between builds)
updatedesktop


Newer Master
-No Update notifications --Pass
Older Master
- Sandbox and Interface updates - Both lead to build 8420 --Pass

Newer Stable
- No Update notifications --Pass
Older Stable
- Sandbox and Interface updates - Interface leads to https://builds.highfidelity.com/HighFidelity-Beta-5654.exe --Pass
>>Sandbox leads to https://builds.highfidelity.com/HighFidelity-Beta-8413.exe  --FAIL

**Comment by birarda** [comment]
@qlimguicheteau You are correct about the SHAs, I have since changed them in the test plan. This is a pass though if that was the only issue (and you saw 4713d8d across the board). Can you confirm?

**Comment by qlimguicheteau** [comment]
On the Older Stable, the Desktop update and the Sandbox update go to different locations. Which is the only thing I can think of that might fail this.

**Comment by birarda** [comment]
@qlimguicheteau right, sorry. That also makes sense - currently the test https://staging.highfidelity.com/builds.xml contains many 0.69.0 entries. The interface code grabs the last one in the file, whereas the sandbox grabs the first. This will not be an issue when the main builds.xml (which this code still points to) has one of each version per file.

Pass?

**Comment by qlimguicheteau** [comment]
If the link was just an organization item, then that looks like a pass.

Pass @conklin94122

**Pull request merged**